src/ForgotYourPasswordDao.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: loic
+ * Date: 11/08/16
+ * Time: 16:22
+ */
+
+namespace Security\Entity\Doctrine;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityRepository;
+use Mouf\Security\Password\Api\ForgotYourPasswordDao as ForgotYourPasswordDaoInterface;
+use Mouf\Security\Password\Exception\EmailNotFoundException;
+use Mouf\Security\UserService\UserInterface;
+
+class ForgotYourPasswordDao extends EntityRepository implements ForgotYourPasswordDaoInterface
+{
+    /**
+     * @var EntityRepository
+     */
+    private $userDao;
+
+    /**
+     * @param EntityRepository $userDao
+     */
+    public function __construct(EntityRepository $userDao)
+    {
+        $this->userDao = $userDao;
+    }
+
+    /**
+     * Sets $token for user whose mail is $email, stores the token in database.
+     * Throws an EmailNotFoundException if the email is not part of the database.
+     *
+     * @param string $email
+     *
+     * @throws \Mouf\Security\Password\Api\EmailNotFoundException
+     */
+    public function setToken(string $email, string $token)
+    {
+        $user = $this->userDao->findOneBy([
```

Comment:
Because you say that this proporty is unique you should use findUniqueBy instead of findOneBy